auv_control_demos/README.md

```diff
@@ -0,0 +1,321 @@
+# AUV Control Demos
+
+This package contains examples that demonstrate how to use the controllers in
+`auv_controllers`. There are two examples. The first shows how to launch just a
+single controller on its own. The second example shows how to chain the
+controllers together to a mock hardware interface for an AUV.
+
+## Example 1: Individual Controller
+
+This example shows how to launch an individual auv controller connect them to a
+mock hardware interface. We launch the integral sliding mode controller which
+takes velocity reference and state interfaces and outputs a wrench command
+interface.
+
+The launch file [individual.launch.py](https://github.com/Robotic-Decision-Making-Lab/auv_controllers/blob/main/auv_control_demos/launch/individual.launch.py)
+contains the code to launch the integral sliding mode controller to a mock
+hardware interface, and that is the launch file we use for this first example.
+
+### Tutorial Steps
+
+1. To test the chainable controller framework, run the following command:
+
+    ```bash
+    ros2 launch auv_control_demos individual.launch.py
+    ```
+
+    The launch file loads and starts the mock hardware and the controller.
+
+2. Check if controllers are running by:
+
+    ```bash
+    ros2 control list_controllers
+    ```
+
+    The output should resemble:
+
+    ```bash
+    integral_sliding_mode_controller[velocity_controllers/IntegralSlidingModeController] active
+    ```
+
+3. Verify if the hardware interfaces are loaded properly by opening another
+  terminal and executing:
+
+    ```bash
+    ros2 control list_hardware_interfaces
+    ```
+
+    The output should resemble:
+
+    ```bash
+    command interfaces
+      integral_sliding_mode_controller/rx/velocity [available] [unclaimed]
+      integral_sliding_mode_controller/ry/velocity [available] [unclaimed]
+      integral_sliding_mode_controller/rz/velocity [available] [unclaimed]
+      integral_sliding_mode_controller/x/velocity [available] [unclaimed]
+      integral_sliding_mode_controller/y/velocity [available] [unclaimed]
+      integral_sliding_mode_controller/z/velocity [available] [unclaimed]
+      rx/effort [available] [claimed]
+      ry/effort [available] [claimed]
+      rz/effort [available] [claimed]
+      x/effort [available] [claimed]
+      y/effort [available] [claimed]
+      z/effort [available] [claimed]
+    state interfaces
+      rx/velocity
+      ry/velocity
+      rz/velocity
+      x/velocity
+      y/velocity
+      z/velocity
+    ```
+
+4. Open another terminal and execute the following command:
+
+    ```bash
+    ros2 topic pub /integral_sliding_mode_controller/system_state geometry_msgs/msg/Twist
+    ```
+
+5. In another terminal, execute the following:
+
+    ```bash
+    ros2 run teleop_twist_keyboard teleop_twist_keyboard --ros-args --remap cmd_vel:=/integral_sliding_mode_controller/reference
```

Comment:
I'm concerned about introducing another dependency here just for a demo. If there is an easier way to demonstrate the control output, I think that would be preferable.

Reply:
We could do something like this
ros2 topic pub /integral_sliding_mode_controller/reference geometry_msgs/msg/Twist "{linear: {x: 0.5, y: 0.0, z: 0.0}, angular: {x: 0.0, y: 0.0, z: 0.2}}"

I think it's not as convenient as using the teleop twist keyboard to be able to put in different twist commands with the tap of a key, but it does cut out the extra dependency.